src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs

```diff
@@ -918,5 +918,34 @@ public static void Main()
                 //         char Moo3(ref char a, ref char b) => r;
                 Diagnostic(ErrorCode.WRN_UnreferencedVar, "Moo3").WithArguments("Moo3").WithLocation(17, 14));
         }
+
+        [Fact]
+        public void NoRefInIndex()
+        {
+            var text = @"
+class C
+{
+    void F(object[] a, int i)
+    {
+        int j;
+        j = a[ref i];    // error 1
+        j = a[out i];    // error 2
+        j = this[ref i]; // error 3
```

Comment:
yes

---
In reply to: [75065315](https://github.com/dotnet/roslyn/pull/13091#discussion_r75065315) [](ancestors = 75065315)